### PR TITLE
PLAT-671: send VDelegatedRequestFinished after retries for correct pending count

### PR DESCRIPTION
**Pull request by aalekseev-ru**
add send, add log in handler for empty latestState case, actualize test

### CODE REVIEW

File: ledger-core/virtual/execute/execute.go [hunk 1]
```diff
@@ -722,6 +722,8 @@ func (s *SMExecute) stepSendOutgoing(ctx smachine.ExecutionContext) smachine.Sta
 		}
 	} else {
 		if s.outgoingSentCounter >= MaxOutgoingSendCount {
+			// TODO when CallSummary will live longer than one pulse it needs to be updated
+			s.sendDelegatedRequestFinished(ctx, nil)
```
Comment (bigbes) [inline ledger-core/virtual/execute/execute.go:726]: I don't like the idea of setting "sendDelegatedRequestFinished" in every place where work can be stopped prematurely :)
I'd prefer to set "ErrorHandler"/"some kind of destructors" somewhere and catch all that situations at once.

Comment (aalekseev-ru) [reply ledger-core/virtual/execute/execute.go:732]: We discuss this with @ruz and decided that this is important for pending tables. Or I can't understand your idea in second sentence.

Comment (bigbes) [reply ledger-core/virtual/execute/execute.go:732]: I mean that part of code (send pending request finished) must be executed on every exit scenario after we've received VDelegatedCallRequest

Comment (bigbes) [reply ledger-core/virtual/execute/execute.go:732]: But that's a big addition, so I guess only if @ruz is OK

Comment (aalekseev-ru) [reply ledger-core/virtual/execute/execute.go:732]: Yeah, we need some kind of destructor for this.

**Pull request merged**